### DISCUSSION TITLE
GitHub Classroom - Is there any feature to sync auto-grading results with LMS?

### DISCUSSION THREAD

**Discussion by Ashenika** [community/community#50890] [GitHub Education]
In the GitHub Classroom, is there any ongoing feature to sync auto-grading results with LMS like Moodle, Canvas etc? 
Or else, any APIs to retrieve auto-grading results to LMS?

Ex: through a Moodle LTI integration to sync roster to GitHub classroom, it will be great if auto-grades also sync to Moodle gradebook.

**Comment by marcel-suter-bzz** [comment]
I wrote an article here [https://github.com/community/Global-Campus-Teachers/discussions/218](https://github.com/community/Global-Campus-Teachers/discussions/218) about my project to sync the grades with moodle assignments.

**Reply by graphics80** [reply]
@Ashenika

**Reply by marcel-suter-bzz** [reply]
@Ashenika I recently started work on a new Moodle plugin to update the students grades for an assignment.
You can check it out at https://github.com/marcel-suter-bzz/assignmentgrade

**Reply by IgnacioPorte** [reply]
Hey @marcel-suter-bzz, I checked your plugin and I think is what I have been looking for, thanks. The link to the article is broken btw.

**Reply by marcel-suter-bzz** [reply]
Thank you for the feedback @IgnacioPorte. Since I wrote the above article, things have moved forward.
The repository and documentation have been moved to a GitHub organization that belongs to our school: 
https://github.com/BZZ-Commons/moodle-mod-externalassignment

The plugin currently runs with Moodle 4.5, 4.5 and 5.0. I'm still testing it with 5.1.

**Comment by sgbaird** [comment]
A non-ideal workaround (at least for Canvas) is to have students self-report their autograded scores using a "New Quizzes" multiple choice question with variable points. For example, if your assignment ranges from 0 to 5 points, you add multiple choice answers for 0, 1, 2, 3, 4, 5, and select the box for varying points by answer, then assign the corresponding points. You can then include a message to students. For example:
> Enter your autograded score between 0 and 5 from the previous assessment. At the end of the course, the self-reported scores will be checked against the autograded scores. If the self-reported score is higher than the autograded score, the autograded score will be used and a 10% penalty will be applied.

<img src=https://github.com/community/community/assets/45469701/7e4b619c-d261-452c-b083-e0e36d797ff8 width=450>

Then, you use the (relatively new) GitHub Classroom CLI to download grades and finagle it into the format the LMS supports for uploading grades (e.g., [Canvas import grades](https://community.canvaslms.com/t5/Instructor-Guide/How-do-I-import-grades-in-the-Gradebook/ta-p/807)). See also the considerations described in https://github.com/community/Global-Campus-Teachers/discussions/275#discussioncomment-7535385 (only available if you're part of the Global Campus for Teachers). tl;dr is that you can run a comparison of the grades locally to check for discrepancies. This would probably be a better discussion topic for the Global Campus for Teachers discussion board.

**Comment by creamcollar1122** [comment]
how can we integrate moodle using lti

**Reply by marcel-suter-bzz** [reply]
I didn't try using LTI. Instead I have a mostly completed work on my Moodle Plugin (https://github.com/BZZ-Commons/moodle-mod-externalassignment   /   https://moodle.org/plugins/mod_externalassignment). With the use of some workflows we can check the source code out students submit in GitHub and send the grading & feedback to moodle.
If you are interested, feel free to reach out to me.

**Reply by creamcollar1122** [reply]
Thank you so much for replying , am trying to integrate github classroom with moodle https://docs.github.com/en/education/manage-coursework-with-github-classroom/teach-with-github-classroom/connect-a-learning-management-system-course-to-a-classroom , following there but ending up in errors saying "Error connecting to GitHub Classroom
Classroom may be misconfigured on your LMS. Contact your LMS administrator.

Error: Missing required information: Context Membership URL" followed exact steps mentioned but no use can anyone guide me on this

**Comment by eliancheng** [comment]
GitHub Classroom does not currently offer a direct feature to sync auto-grading results with Learning Management Systems (LMS) like Moodle or Canvas. But my thinking is same to you.
